upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
```

Comment:
While the use of an IIFE here is clever to start execution, consider refactoring to make the function more clearly structured, perhaps by extracting the chunk upload logic into a named function. This aids readability and maintainability.